### PR TITLE
Add: prettier, linter, lint-staged and husky

### LINKED ISSUE #1: Add/configure prettier and lint

### LINKED ISSUE #4: Add husky with lint staged 🐶

### PR DESCRIPTION
**Pull request by mizek1**
fix #1
fix #4 

I ran prettier and linter commands, fixed some issues related to linter warnings and errors and changed tests commands to run in sequence.

### CODE REVIEW

**Review by MarcosOliveir4 — APPROVED** [review]
Depois sobe a pasta .vscode com os plugins recomendados, também pode ser criado um arquivo dentro dela chamado settings.json colocando a configuração de ao salvar formatar: "editor.formatOnSave": true. 👍

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+  "env": {
+    "es2021": true,
+    "browser": true
+  },
+  "rules": {
+    "@typescript-eslint/no-explicit-any": "error",
+    "no-console": 1
```
Comment (iurynogueira) [inline .eslintrc:9]: We can keep pattern in values, I prefer string values. What do you think?

File: package.json [hunk 2]
```diff
@@ -9,9 +9,11 @@
     "start": "ng serve",
     "build": "ng build",
     "watch": "ng build --watch --configuration development",
-    "test": "jest",
-    "test:coverage": "jest --coverage",
-    "test:watch": "jest --watchAll"
+    "test": "jest -i",
+    "test:coverage": "jest -i --coverage",
+    "test:watch": "jest -i --watchAll",
```
Comment (iurynogueira) [inline package.json:14]: Nice -i!

**Review by iurynogueira — CHANGES_REQUESTED** [review]
Good! This is was fast!

File: .eslintrc (hunk 1 above)
Comment (mizek1) [reply .eslintrc:9]: This config doesn't accept strings as values, I don't know why, prob because it's an enum.

Comment (iurynogueira) [reply .eslintrc:9]: Sad 😢

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**